compiler/src/main/java/org/qbicc/graph/Node.java

```diff
@@ -283,10 +301,53 @@ public Value visitUnknown(final Copier param, final Value node) {
             }
 
             public Value visitAny(Copier copier, Literal literal) {
-                // literals remain unchanged by default
+                // literals without value dependencies remain unchanged by default
+                Assert.assertTrue(literal.getValueDependencyCount() == 0);
                 return literal;
             }
 
+            public Value visit(Copier copier, ArrayLiteral literal) {
+                List<Literal> values = copier.copyLiterals(literal.getValues());
+                return new ArrayLiteral(literal.getType(), values);
```

Comment:
```suggestion
                return copier.getBlockBuilder().getLiteralFactory().literalOf(literal.getType(), values);
```

This approach would avoid needing to make the ctors public, and would allow us to collapse duplicates if we ever added a cache for it. Similar comment for the other ones.

Reply:
Good suggestion.  Fixed.